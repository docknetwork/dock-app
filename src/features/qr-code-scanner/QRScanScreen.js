import Clipboard from '@react-native-community/clipboard';
import React from 'react';
import {
  Dimensions,
  StyleSheet,
  Text,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import {RNCamera} from 'react-native-camera';
import QRCodeScanner from 'react-native-qrcode-scanner';
import {Theme} from 'src/design-system';
import styled from 'styled-components/native';
import {navigateBack} from '../../core/navigation';
import {Colors} from '../../theme/colors';

const Container = styled.View`
  flex: 1;
  background-color: ${Colors.darkBlue};
`;

const Header = styled.View`
  height: 80px;
  width: 100%;
  padding-top: 10px;
`;

const Title = styled.Text`
  color: ${Colors.white};
  font-size: 18px;
  line-height: 24px;
  width: 100%;
  text-align: center;
`;

const IconContainer = styled.View`
  position: absolute;
  top: 0px;
  left: 10px;
  padding: 20px;
  z-index: 99999;
`;

const Footer = styled.View`
  display: flex;
  justify-content: center;
  background: #2e3945;
  border-radius: 4px;
  margin: 20px;
  height: 93px;
`;

const Description = styled.Text`
  color: ${Colors.white};
  font-size: 18px;
  text-align: center;
`;

const SmallDescription = styled.Text`
  color: ${Colors.white};
  font-size: 13px;
  text-align: center;
`;

const Body = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
`;

const Wrapper = styled.SafeAreaView`
  display: flex;
  flex-direction: column;
  flex: 1;
`;

const QRCodeContainer = styled.View`
  position: absolute;
  width: 100%;
  height: 100%;
`;

export function QRScanScreen({route}) {
  const {onData} = route.params;

  return (
    <Container>
      <QRCodeContainer>
        <QRCodeScanner
          style={styles.scanner}
          reactivate={true}
          reactivateTimeout={10000}
          flashMode={RNCamera.Constants.FlashMode.off}
          cameraStyle={{height: Dimensions.get('window').height}}
          topViewStyle={{
            height: 0,
            flex: 0,
          }}
          bottomViewStyle={{
            height: 0,
            flex: 0,
          }}
          onRead={event => onData(event.data)}
        />
      </QRCodeContainer>
      <Wrapper>
        <Header>
          <Title>Scan QR Code</Title>
          <IconContainer>
            <TouchableWithoutFeedback
              onPress={navigateBack}
              style={{padding: 20}}>
              <Text style={{color: Theme.colors.textHighlighted}}>Back</Text>
            </TouchableWithoutFeedback>
          </IconContainer>
        </Header>
        <Body>
          <View style={{width: 240, height: 230}}>
            <View style={[styles.frame, styles.frameLeftTop]} />
            <View style={[styles.frame, styles.frameRightTop]} />
            <View style={[styles.frame, styles.frameLeftBottom]} />
            <View style={[styles.frame, styles.frameRightBottom]} />
          </View>
        </Body>
        <TouchableWithoutFeedback
          onPress={async () => {
            const text = await Clipboard.getString();
            onData(text);
          }}>
          <Footer>
            <Description>Scan the QR Code</Description>
            <SmallDescription>Place the code inside the box</SmallDescription>
          </Footer>
        </TouchableWithoutFeedback>
      </Wrapper>
    </Container>
  );
}

const styles = StyleSheet.create({
  centerText: {
    flex: 1,
    fontSize: 18,
    padding: 32,
    color: '#777',
  },
  textBold: {
    fontWeight: '500',
    color: '#000',
  },
  buttonText: {
    fontSize: 21,
    color: 'rgb(0,122,255)',
  },
  buttonTouchable: {
    padding: 16,
  },
  scanner: {
    flex: 1,
  },
  frame: {
    width: 60,
    height: 60,
    borderStyle: 'solid',
    borderColor: '#00C0D9',
    borderLeftWidth: 12,
    borderTopWidth: 12,
    position: 'absolute',
  },
  frameLeftTop: {
    left: 0,
    top: 0,
  },
  frameRightTop: {
    right: 0,
    top: 0,
  },
  frameLeftBottom: {
    left: 0,
    bottom: 0,
  },
  frameRightBottom: {
    right: 0,
    bottom: 0,
  },
});
