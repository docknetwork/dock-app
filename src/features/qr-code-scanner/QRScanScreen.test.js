import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {QRScanScreen} from './QRScanScreen';

const mockStore = configureMockStore();

describe('QRScanScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<QRScanScreen route={{}} />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
