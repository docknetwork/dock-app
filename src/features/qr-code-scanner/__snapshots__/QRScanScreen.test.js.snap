// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRScanScreen should render correctly 1`] = `
<StyledNativeComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "inlineStyle": InlineStyle {
        "rules": Array [
          "
  flex: 1;
  background-color: ",
          "#013c73",
          ";
",
        ],
      },
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "StyledNativeComponent",
      "target": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <Styled(View) />
  <Styled(Component)>
    <Styled(View)>
      <Styled(Text)
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Satoshi",
          }
        }
      >
        Scan QR Code
      </Styled(Text)>
      <Styled(View)>
        <TouchableWithoutFeedback
          onPress={[Function]}
          style={
            Object {
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "Satoshi",
              }
            }
          >
            Back
          </Text>
        </TouchableWithoutFeedback>
      </Styled(View)>
    </Styled(View)>
    <Styled(View)>
      <View
        style={
          Object {
            "height": 230,
            "width": 240,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#00C0D9",
                "borderStyle": "solid",
                "height": 60,
                "position": "absolute",
                "width": 60,
              },
              Object {
                "borderLeftWidth": 12,
                "borderTopWidth": 12,
                "left": 0,
                "top": 0,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderColor": "#00C0D9",
                "borderStyle": "solid",
                "height": 60,
                "position": "absolute",
                "width": 60,
              },
              Object {
                "borderRightWidth": 12,
                "borderTopWidth": 12,
                "right": 0,
                "top": 0,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderColor": "#00C0D9",
                "borderStyle": "solid",
                "height": 60,
                "position": "absolute",
                "width": 60,
              },
              Object {
                "borderBottomWidth": 12,
                "borderLeftWidth": 12,
                "bottom": 0,
                "left": 0,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderColor": "#00C0D9",
                "borderStyle": "solid",
                "height": 60,
                "position": "absolute",
                "width": 60,
              },
              Object {
                "borderBottomWidth": 12,
                "borderRightWidth": 12,
                "bottom": 0,
                "right": 0,
              },
            ]
          }
        />
      </View>
    </Styled(View)>
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <Styled(View)>
        <Styled(Text)>
          Scan the QR Code
        </Styled(Text)>
        <Styled(Text)>
          Place the code inside the box
        </Styled(Text)>
      </Styled(View)>
    </TouchableWithoutFeedback>
  </Styled(Component)>
</StyledNativeComponent>
`;
